File: src/test_db.js
const { MongoClient } = require('mongodb');

// FerretDB listens on the standard Mongo port 27017
const url = 'mongodb://127.0.0.1:27017';
const client = new MongoClient(url);

async function main(){
	await client.connect();
	console.log('Successfully connected to FerretDB!');


        const db = client.db('pi_database');
        const collection = db.collection('logs');

        // Insert a test document
        await collection.insertOne({ 
            device: "Raspberry Pi", 
            status: "active", 
            timestamp: new Date() 
        });
	
	console.log("===================== DONE ===================\n")
}//main


main()

File: src/python/test_db.py
from pymongo import MongoClient

# 1. Setup the connection string
# Use '127.0.0.1' instead of 'localhost' to avoid IPv6 resolution delays on Ubuntu
uri = "mongodb://127.0.0.1:27017/"

# 2. Create the client
client = MongoClient(uri)

try:
    # 3. Send a ping to confirm a successful connection
    client.admin.command('ping')
    print("✅ Successfully connected to MongoDB/FerretDB!")
    
    # 4. Access a database and collection
    db = client["sensor_db"]
    collection = db["readings"]

    # 5. Insert and find a document
    collection.insert_one({"sensor": "RPi_Temp", "value": 42.5})
    print("Found document:", collection.find_one({"sensor": "RPi_Temp"}))

except Exception as e:
    print(f"❌ Connection failed: {e}")

finally:
    client.close()
    print("====================== DONE ========================")